File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let mxsum=-Infinity,sum=0;
    for(let num of nums){
        sum+=num;
        mxsum=Math.max(mxsum,sum);
        if(sum<0)sum=0;
    }
    return mxsum;
};

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int mxsum=INT_MIN,sum=0;
        if(nums.size()==1)return nums[0];
        for(int i=0;i<nums.size();i++){
           sum+=nums[i];
           mxsum=max(mxsum,sum);
           if(sum<0){
            sum=0;
           }
        }
        return mxsum;
    }
};